cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -490,19 +490,23 @@ func StartControllers(controllers map[string]InitFunc, s *options.CMServer, root
 	}
 
 	if ctx.IsControllerEnabled(serviceControllerName) {
-		serviceController, err := servicecontroller.New(
-			cloud,
-			clientBuilder.ClientOrDie("service-controller"),
-			sharedInformers.Core().V1().Services(),
-			sharedInformers.Core().V1().Nodes(),
-			s.ClusterName,
-		)
-		if err != nil {
-			glog.Errorf("Failed to start service controller: %v", err)
+		if cloud != nil {
```

Comment:
Sorry, didn't notice that we tolerant this error instead of return. Yes then, please check cloud==nil first.